src/rgw/Makefile.am

```diff
@@ -93,6 +93,11 @@ librgw_la_SOURCES = \
 
 if WITH_OPENLDAP
 librgw_la_SOURCES += rgw/rgw_ldap.cc
+
+if FREEBSD
```

Comment:
nit, could you move this block down to https://github.com/ceph/ceph/pull/8298/commits/886d907bd4930d7d1f1626a9f3e1611c3d8901d3#diff-d8612c96429ff7fbf2de590a08bca6dbR123, so all LIBRGW_DEPS related code can be grouped together?